src/script/lua_api/l_object.cpp

```diff
@@ -1936,6 +1940,18 @@ int ObjectRef::l_get_sky(lua_State *L)
 
 	push_sky_color(L, skybox_params);
 	lua_setfield(L, -2, "sky_color");
+
+	lua_newtable(L); // fog
+	if (skybox_params.fog_distance >= 0) {
+		lua_pushinteger(L, skybox_params.fog_distance);
+		lua_setfield(L, -2, "fog_distance");
+	}
+	if (skybox_params.fog_start >= 0) {
+		lua_pushnumber(L, skybox_params.fog_start);
+		lua_setfield(L, -2, "fog_start");
+	}
+	lua_setfield(L, -2, "fog");
```

Comment:
I'm still for pushing `-1` here. After all, all the other params are also not `nil` if they're the default value.
https://github.com/minetest/minetest/pull/13448/files#r1199254966